geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/ListGatewayCommand.java

```diff
@@ -53,9 +53,51 @@ public ResultModel listGateway(
           help = CliStrings.LIST_GATEWAY__MEMBER__HELP) String[] onMember,
       @CliOption(key = {CliStrings.GROUP, CliStrings.GROUPS},
           optionContext = ConverterHint.MEMBERGROUP,
-          help = CliStrings.LIST_GATEWAY__GROUP__HELP) String[] onGroup)
+          help = CliStrings.LIST_GATEWAY__GROUP__HELP) String[] onGroup,
+      @CliOption(key = {CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY},
+          specifiedDefaultValue = CliStrings.LIST_GATEWAY__SHOW_PARAM_TRUE,
+          unspecifiedDefaultValue = CliStrings.LIST_GATEWAY__SHOW_PARAM_EMPTY,
+          help = CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY) String showReceiversOnly,
+      @CliOption(key = {CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY},
+          specifiedDefaultValue = CliStrings.LIST_GATEWAY__SHOW_PARAM_TRUE,
+          unspecifiedDefaultValue = CliStrings.LIST_GATEWAY__SHOW_PARAM_EMPTY,
+          help = CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY__HELP) String showSendersOnly)
```

Comment:
Both parameters should be of type `boolean`, and you can achieve the same outcome directly through `spring-shell` features:

```
@CliOption(key = CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY, specifiedDefaultValue = "true", unspecifiedDefaultValue = "false", help = CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY) boolean showOnlySenders

@CliOption(key = CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY, specifiedDefaultValue = "true", unspecifiedDefaultValue = "false", help = CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY) boolean showOnlyReceivers
```

Spring shell will automatically populate the correct value for the `boolean` flags when parsing the user input, as below:

| Command  | showOnlyReceivers | showOnlySenders| 
| ------------- | ------------- | ------------- |
| list gateways  | false  | false |
| list gateways --senders-only  | false | true |
| list gateways --senders-only=true  | false  | true |
| list gateways --senders-only=false  | false  | false |
| list gateways --receivers-only  | true  | false |
| list gateways --receivers-only=true  | true  | false |
| list gateways --receivers-only=false  | false  | false |

Reply:
> Both parameters should be of type `boolean`, and you can achieve the same outcome directly through `spring-shell` features:
> 
> ```
> @CliOption(key = CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY, specifiedDefaultValue = "true", unspecifiedDefaultValue = "false", help = CliStrings.LIST_GATEWAY__SHOW_SENDERS_ONLY) boolean showOnlySenders
> 
> @CliOption(key = CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY, specifiedDefaultValue = "true", unspecifiedDefaultValue = "false", help = CliStrings.LIST_GATEWAY__SHOW_RECEIVERS_ONLY) boolean showOnlyReceivers
> ```
> 
> Spring shell will automatically populate the correct value for the `boolean` flags when parsing the user input, as below:
> Command 	showOnlyReceivers 	showOnlySenders
> list gateways 	false 	false
> list gateways --senders-only 	false 	true
> list gateways --senders-only=true 	false 	true
> list gateways --senders-only=false 	false 	false
> list gateways --receivers-only 	true 	false
> list gateways --receivers-only=true 	true 	false
> list gateways --receivers-only=false 	false 	false

And how do you difference between `list gateways --receivers-only --senders-only=false` and `list gateways --senders-only=false`? In both cases, `showOnlyReceivers=true` and `showOnlySenders=false` but the first one should return error and the second one should work.

Reply:
I have just read your other comment about interceptors. If using an interceptor Im able to return error when `--senders-only`& `--receivers-only` are used together, then I could go back to use a boolean.